app/api/ada/adaRedemption.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import { getAddressFromRedemptionKey, getRedemptionSignedTransaction } from './lib/cardanoCrypto/cryptoRedemption';
+import bs58 from 'bs58';
+import { getUTXOsForAddresses, sendTx } from './lib/yoroi-backend-api';
+import { decryptRegularVend } from './lib/decrypt';
+import { getReceiverAddress } from './adaAddress';
+import { RedemptionKeyAlreadyUsedError } from './errors';
+import BigNumber from 'bignumber.js';
+
+export type RedeemAdaParams = {
+  redemptionCode: string,
+  walletId: string,
+  accountIndex: number
+};
+
+export type RedeemPaperVendedAdaParams = {
+  redemptionCode: string,
+  walletId: string,
+  accountIndex: number,
+  mnemonics: Array<string>,
+};
+
+export async function redeemAda(
```

Comment:
Is it okay if we add comments explaining the expected behaviour?

Reply:
If there is no documentation on this anywhere then maybe comments would be good but if this would take time then don't worry so much about it. We won't be editing this part of the logic so it's not a big deal if it's a black box.